File: 1761-minimum-degree-of-a-connected-trio-in-a-graph/1761-minimum-degree-of-a-connected-trio-in-a-graph.py
class Solution:
    def minTrioDegree(self, n: int, edges: List[List[int]]) -> int:
        deg = [0]*n # For Storing Degrees of nodes
        G = [[False]*n for _ in range(n)] # Adjacency Matrix as we need to check connectivity between any 2 nodes in O(1)
        
        # Filling degree list and connectivity Matrix for faster lookup
        for u,v in edges:
            u, v = u-1, v-1
            G[u][v] = G[v][u] = True
            deg[u] += 1
            deg[v] += 1
    
        # Try for all combinations of 3 Nodes and check if they form a trio 
        # i.e. for any 3 nodes i, j, k in trio G[i][j] = G[j][k] = G[i][k] = True
        ans = math.inf
        for i in range(n):
            for j in range(i+1,n):
                if G[i][j]:
                    for k in range(j+1,n):
                        if G[i][k] and G[j][k]:
                            ans = min(ans, deg[i] + deg[j] + deg[k] - 6)
        
        return ans if ans < math.inf else -1
        

File: 1761-minimum-degree-of-a-connected-trio-in-a-graph/1761-minimum-degree-of-a-connected-trio-in-a-graph.java
class Solution {
    public int minTrioDegree(int N, int[][] edges) {
        boolean[][] connected = new boolean[N][N];
        int[] deg = new int[N];
        
        for(int i = 0 ; i < edges.length ; ++i){
            int u = edges[i][0] - 1;
            int v = edges[i][1] - 1;
            connected[u][v] = true;
            connected[v][u] = true;
            deg[u]++;
            deg[v]++;
        }
        
        int total = 100000000;
        for(int i=0 ; i<N ; ++i){
            for(int j = i+1 ; j < N ; ++j){
                if(connected[i][j]){
                    for(int k = j+1 ; k < N ; ++k){
                        if(connected[j][k] && connected[i][k]){
                            total = Math.min(total, deg[i] + deg[j] + deg[k] - 6);
                        }
                    }
                }
            }
        }
        
        if(total == 100000000){
            return -1;
        }
        
        return total;
       
    }
}